src/main/java/org/spongepowered/api/text/dialogue/DialogueArchetype.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.dialogue;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataQuery;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.text.channel.MessageChannel;
+import org.spongepowered.api.util.ResettableBuilder;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Collection;
+
+/**
+ * Represents a structure for a back-and-forth communication between one or
+ * more {@link Speaker}s and one or more plugiins.
+ */
+public interface DialogueArchetype {
+
+    /**
+     * Creates a new {@link Builder} to build a new {@link DialogueArchetype}.
+     *
+     * @return The newly created builder
+     */
+    static Builder builder() {
+        return Sponge.getRegistry().createBuilder(Builder.class);
+    }
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Speaker... speakers);
+
+    /**
+     * Creates a new {@link Dialogue} for the specified {@link Speaker}(s).
+     *
+     * @param speakers The speakers to send the initial {@link Question} to
+     * @return The newly created {@link Dialogue}
+     * @throws IllegalArgumentException If {@code speakers} is empty
+     */
+    Dialogue start(Iterable<? extends Speaker> speakers);
+
+    interface Builder extends ResettableBuilder<DialogueArchetype, Builder> {
```

Comment:
Maybe its just my superficial impression, but shouldn't be the `Builder` definition be the last member in a class? (At least that is what the `Text` classes do).